xbmc/video/Bookmark.h

```diff
@@ -38,20 +38,24 @@ class CBookmark
    */
   bool IsPartWay() const;
 
+  bool IsMediaBookmark() const;
+  static bool IsMediaBookmark(int t);
+
   double timeInSeconds;
   double totalTimeInSeconds;
   long partNumber;
   std::string thumbNailImage;
   std::string playerState;
   std::string player;
-  long seasonNumber;
-  long episodeNumber;
+  std::string title;
 
   enum EType
   {
     STANDARD = 0,
     RESUME = 1,
-    EPISODE = 2
+    EPISODE = 2,
```

Comment:
My first work was only with a "MEDIA" type and nothing to actually say if it was movie, episode or musicvideo. It works and the code is cleaner but in a couple of cases like in ClearBookMarkOfFile you don't know if the file is an episode, a movie or a musicinfo so you need to do 3 queries to update the media table instead of one. 
I could store the media type in a separate column but I don't see the advantage. If there's a new type you still need to extend it everywhere and it's harder to pass arguments because you don't only have anymore the bookmark type but also the bookmark media type